pkgs/build-support/fetchbittorrent/default.nix

```
@@ -0,0 +1,59 @@
+{ lib, runCommand, transmission-withoutSystemd, rqbit, writeScript, formats, cacert, rsync }:
+let
+  urlRegexp = ''.*xt=urn:bt[im]h:([^&]{64}|[^&]{40}).*'';
+in
+{ url
+, name ?
+  if (builtins.match urlRegexp url) == null then
+    "bittorrent"
+  else
+    "bittorrent-" + builtins.head (builtins.match urlRegexp url)
+, config ? if (backend == "transmission") then { } else throw "json config for configuring fetchFromBitorrent only works with the transmission backend"
+, hash
+, backend ? "transmission"
+, recursiveHash ? true
+, postFetch ? ""
+, postUnpack ? ""
+}:
+let
+  afterSuccess = writeScript "kill.sh" ''
+    ${postUnpack}
+    # Flatten the directory, so that only the torrent contents are in $out, not
+    # the folder name
+    mv -v $downloadedDirectory/*/* $out
```

This doesn't work for dot files.

```suggestion
    shopt -s dotglob
    mv -v $downloadedDirectory/*/* $out
```

Haven't tested this.